### ISSUE TITLE
Unnecessary `unwrap_unchecked`

### ISSUE THREAD

**Issue by Nugine** [rust-lang/rust-clippy#11108]
### What it does

Detects:
```rust
std::str::from_utf8(s).unwrap_unchecked()
```

Suggests:
```rust
std::str::from_utf8_unchecked(s)
```

### Advantage

- Skip the validation since the code doesn't need it
- Suggest the more idiomatic method

### Example

Detects:
```rust
std::str::from_utf8(s).unwrap_unchecked()
```

Suggests:
```rust
std::str::from_utf8_unchecked(s)
```

**Comment by Centri3** [comment]
Would this detect all libstd functions with an `_unchecked` alternative followed by `unwrap_unchecked`? Could be a neat lint.

**Comment by Nugine** [comment]
> Would this detect all libstd functions with an `_unchecked` alternative followed by `unwrap_unchecked` (or perhaps even more?)? Could be a neat lint.

Yes. It would be useful when someone is not familiar with the unsafe methods in libstd.
